code/modules/lighting/light_effect_cast.dm

```diff
@@ -0,0 +1,303 @@
+		num = 2
+
+	//due to only having one set of shadow templates, we need to rotate and flip them for up to 8 different directions
+	//first check is to see if we will need to "rotate" the shadow template
+	var/xy_swap = 0
+	if(abs(x_offset) > abs(y_offset))
+		xy_swap = 1
+
+	var/shadowoffset = WORLD_ICON_SIZE/2 + (WORLD_ICON_SIZE*light_range)
+
+	// An explicit call to file() is easily 1000 times as expensive than this construct, so... yeah.
+	// Setting icon explicitly allows us to use byond rsc instead of fetching the file everytime.
+	// The downside is, of course, that you need to cover all the cases in your switch.
+	var/shadowicon
+	switch(light_range)
+		if(2)
+			if(num == 1)
+				shadowicon = 'icons/lighting/light_range_2_shadows1.dmi'
+			else
+				shadowicon = 'icons/lighting/light_range_2_shadows2.dmi'
+		if(3)
+			if(num == 1)
+				shadowicon = 'icons/lighting/light_range_3_shadows1.dmi'
+			else
+				shadowicon = 'icons/lighting/light_range_3_shadows2.dmi'
+		if(4)
+			if(num == 1)
+				shadowicon = 'icons/lighting/light_range_4_shadows1.dmi'
+			else
+				shadowicon = 'icons/lighting/light_range_4_shadows2.dmi'
+		if(5)
+			if(num == 1)
+				shadowicon = 'icons/lighting/light_range_5_shadows1.dmi'
+			else
+				shadowicon = 'icons/lighting/light_range_5_shadows2.dmi'
+
+	var/image/I = image(shadowicon)
+
+	//due to the way the offsets are named, we can just swap the x and y offsets to "rotate" the icon state
+	if(xy_swap)
+		I.icon_state = "[abs(y_offset)]_[abs(x_offset)]"
+	else
+		I.icon_state = "[abs(x_offset)]_[abs(y_offset)]"
+
+	var/matrix/M = matrix()
+
+	//TODO: rewrite this comment:
+	//using scale to flip the shadow template if needed
+	//horizontal (x) flip is easy, we just check if the offset is negative
+	//vertical (y) flip is a little harder, if the shadow will be rotated we need to flip if the offset is positive,
+	// but if it wont be rotated then we just check if its negative to flip (like the x flip)
+	var/x_flip
+	var/y_flip
+	if(xy_swap)
+		x_flip = y_offset > 0 ? -1 : 1
+		y_flip = x_offset < 0 ? -1 : 1
+	else
+		x_flip = x_offset < 0 ? -1 : 1
+		y_flip = y_offset < 0 ? -1 : 1
+
+	M.Scale(x_flip, y_flip)
+
+	//here we do the actual rotate if needed
+	if(xy_swap)
+		M.Turn(90)
+
+	//warning: you are approaching shitcode (this is where we move the shadow to the correct quadrant based on its rotation and flipping)
+	//shadows are only as big as a quarter or half of the light for optimization
+
+	//please for the love of god change this if there's a better way
+
+	if(num == 1)
+		if((x_flip == 1 && y_flip == 1 && xy_swap == 0) || (x_flip == -1 && y_flip == 1 && xy_swap == 1))
+			M.Translate(shadowoffset, shadowoffset)
+		else if((x_flip == 1 && y_flip == -1 && xy_swap == 0) || (x_flip == 1 && y_flip == 1 && xy_swap == 1))
+			M.Translate(shadowoffset, 0)
+		else if((xy_swap == 0 && x_flip == -y_flip) || (xy_swap == 1 && x_flip == -1 && y_flip == -1))
```

Comment:
The first condition (`(xy_swap == 0 && x_flip == -y_flip)`) in this line will never be true if this line is executed. If that condition is true, then the previous `else if()` would have executed instead.